restapi/src/main/java/io/stargate/web/resources/Db.java

```diff
@@ -136,4 +145,42 @@ public DocumentDB getDocDataStoreForToken(String token, int pageSize, ByteBuffer
         authenticationSubject,
         getAuthorizationService());
   }
+
+  static class TokenAndHeaders {
+    private static final String TENANT_ID_HEADER = "tenant_id";
+    private static final String HOST_HEADER = "Host";
+    private final String token;
+    private final Map<String, String> headers;
+
+    static TokenAndHeaders create(String token, Map<String, String> headers) {
+      return new TokenAndHeaders(token, filterHeaders(headers));
+    }
+
+    private static Map<String, String> filterHeaders(Map<String, String> headers) {
```

Comment:
Why are we filtering headers? We need to make this work for everyone.

Reply:
this is for building the cache key. We want the cache to be per `token`, `Host`, and `tenant_id`. 
It was discussed here:
https://github.com/stargate/stargate/pull/581#discussion_r559756983

Reply:
Implemented in:
https://github.com/stargate/stargate/pull/581/commits/05d379881dcf18262d3d07eaf21ccb33dad3960c

Reply:
The host contains the tenantID so we don't need both, only the host header.

Reply:
For `cql` it will contain tenant_id. So it can be one or another.

Reply:
This is only for the restapi module so it should be dealing with just the host header. We should avoid referencing the tenantID as much as possible to avoid confusion for anyone trying to modify this code.

Reply:
good point, done